pycbc/waveform/waveform.py

```diff
@@ -122,6 +122,15 @@ def _check_lal_pars(p):
         lalsimulation.SimInspiralWaveformParamsInsertFrameAxis(lal_pars, p['frame_axis'])
     if p['side_bands']:
         lalsimulation.SimInspiralWaveformParamsInsertSideband(lal_pars, p['side_bands'])
+    if p['mode_array']:
+        try:
+            ma = lalsimulation.SimInspiralCreateModeArray()
+            for l,m in p['mode_array']:
+                lalsimulation.SimInspiralModeArrayActivateMode(ma, l, m)
+            lalsimulation.SimInspiralWaveformParamsInsertModeArray(lal_pars, ma)
+        except AttributeError:
```

Comment:
I want this to fail here if the user provides a ModeArray but the lalsimulation being used doesn't have it (so this fails). This would stop someone being confused if they are *expecting* this to work, it appears to, but is actually doing something different.

Reply:
How do you suggest I implement a failure here?

simply something like

```
sys.exit()
```

I also think that the travis tests will be marked as failed and thus won't be able to merge this PR because the travis system is using an older lalsimulation. So we would have to force merge this PR. Are you still OK with this?

Reply:
Hi @Cyberface, sorry for previous abrupt comment. I think the following set of behaviour should happen:

 * Old version of lalsuite + mode_array not given: Run as normal
 * New version of lalsuite + mode_array given: Use mode array, YAY!
 * New version of lalsuite + mode array not given: Run as normal
 * Old version of lalsuite + mode_array given: FAIL

The Travis tests do not check with mode_array being provided, so should not fail. What am I missing?

You can raise a failure just by removing the try/except stuff. Don't try to set mode-array stuff if the user has not set a non-default value.